# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [6]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v ggplot2 2.2.1     v purrr   0.2.4
v tibble  1.4.1     v dplyr   0.7.4
v tidyr   0.7.2     v stringr 1.2.0
v readr   1.1.1     v forcats 0.2.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [7]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2         V3         V4        V5        V6        V7        
1 0.05973998 0.06666556 0.31421311 0.9596913 0.5114395 0.2733971 0.95668943
2 0.39921438 0.91077715 0.01460446 0.8119931 0.7975737 0.5623284 0.08047719
  V8         V9        V10        V11       V12       V13       V14       
1 0.01182025 0.7517132 0.02615967 0.8393612 0.0250788 0.5229996 0.07612415
2 0.76610337 0.9446619 0.75059387 0.9153833 0.5035952 0.5423270 0.57160338
  V15       V16        V17       V18       V19       V20      
1 0.8394540 0.08554995 0.1229474 0.4593851 0.2718864 0.5503199
2 0.8789465 0.93442413 0.6314634 0.3786453 0.4396742 0.7433768

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [8]:
df2 <- df1 %>%
  gather(key,value)

head(df2)

key value     
1 V1  0.05973998
2 V1  0.39921438
3 V2  0.06666556
4 V2  0.91077715
5 V3  0.31421311
6 V3  0.01460446

## Assignment 2

We will generate some data:

In [9]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [10]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [12]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook